### PR TITLE
WIP: add an input_datetime

### PR DESCRIPTION
**Pull request by tinloaf**
This accompanies the PR adding an 'input_datetime': See https://github.com/home-assistant/home-assistant/pull/9313

I'm not yet really sure when to store a value as string (as in case of the date, since the date picker likes its data as string) and when in some other form (as for the time picker, which likes hours and minutes as numbers). Also, my JavaScript and HTML are probably horrible. I welcome all suggestions for improvement!

### CODE REVIEW

File: src/more-infos/more-info-input_datetime.html [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      </div>
+      <div hidden$='[[doesNotHaveTime(stateObj)]]'>
+        <paper-time-input hour="{{selectedHour}}" min="{{selectedMinute}}" format="24"></paper-time-input>
+      </div>
+    </div>
+  </template>
+</dom-module>
+
+<script>
+Polymer({
```
Comment (balloob) [inline src/more-infos/more-info-input_datetime.html]: Please convert this to a Polymer 2 class.

File: src/util/hass-util.html [hunk 2]
```diff
@@ -494,6 +494,21 @@
         case 'power':
         default:
       }
+    } else if (window.hassUtil.computeDomain(stateObj) === 'input_datetime') {
```
Comment (balloob) [inline src/util/hass-util.html]: Let's cache the domain in a local variable instead of computing it twice

**Review by balloob — CHANGES_REQUESTED** [review]

File: src/more-infos/more-info-input_datetime.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  is: 'more-info-input_datetime',
+
+  properties: {
+    hass: {
```
Comment (andrey-git) [inline src/more-infos/more-info-input_datetime.html]: For properties with just the type - use the short syntax:
```
hass: Object
```

File: src/more-infos/more-info-input_datetime.html [hunk 4]
```diff
@@ -0,0 +1,130 @@
+<link rel="import" href="../../bower_components/polymer/polymer.html">
+<link rel="import" href="../../bower_components/iron-flex-layout/iron-flex-layout-classes.html">
+<link rel="import" href="../components/ha-relative-time.html">
+<link rel="import" href="../../bower_components/vaadin-date-picker/vaadin-date-picker.html">
+<link rel="import" href="../../bower_components/paper-time-input/paper-time-input.html">
+
+<dom-module id="more-info-input_datetime">
+  <template>
+    <style is="custom-style" include="iron-flex iron-flex-alignment"></style>
+    <div class$='[[computeClassNames(stateObj)]]'>
```
Comment (andrey-git) [inline src/more-infos/more-info-input_datetime.html]: You are not styling anything according to those classes

File: src/more-infos/more-info-input_datetime.html [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      dayFiller = '0';
+    } else {
+      dayFiller = '';
+    }
+
+    return stateObj.attributes.year + '-' + monthFiller + stateObj.attributes.month + '-' + dayFiller + stateObj.attributes.day;
+  },
+
+  dateTimeChanged: function () {
+    var minuteFiller;
```
Comment (andrey-git) [inline src/more-infos/more-info-input_datetime.html]: ES6 style: let / const instead of var

**Comment by tinloaf** [comment]
I have converted the element to a Polymer 2 class now, however there are some things of which I'm not sure whether I solved them 'correctly':

- The is_ready flag seems to be a hack to me, but as far as I can see the lifecycle of a polymer 2 class does not allow to check whether an 'attribute changed' callback fires because of actual user interaction or because initial values are being set. Likewise, at least the vaadin-date-picker element does not seem to have a 'on_change' attribute.

-  Sometimes it seems like I cannot open the date picker dialog or change the time while the history is still loading. Both the date picker and the time picker are just not responding to user interaction while the circle still spins…

**Comment by balloob** [comment]
Why not listen to `value-changed` on the date picker? [Date picker docs](https://www.webcomponents.org/element/vaadin/vaadin-date-picker/elements/vaadin-date-picker#events)

**Comment by tinloaf** [comment]
@balloob it low listens to value-changed. I didn't see the value-changed event for the date picker, because it's not listed under the events. :-) Thanks for the hint!

However, there is no such event for the time picker, thus I'm still using the value observers for that.

**Comment by balloob** [comment]
Nice :dolphin: 🌮

**Review by balloob — APPROVED** [review]

**Pull request merged**